docs(sdk): update contributing doc to include updated setup (#7203)
wandb/wandb: CONTRIBUTING.md

We test the library code against multiple `python` versions
and use [`pyenv`](https://github.com/pyenv/pyenv) to manage those. Install `pyenv` by running

> following the instruction in [here](https://github.com/pyenv/pyenv?tab=readme-ov-file#getting-pyenv).

Reviewer: ```diff
- Install `pyenv` by running following the instruction in [...]
+ Install `pyenv` by following the instructions at [...]
```

You would also likely want to setup pyenv-virtualenv to manage multiple environement.
For more details see: [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv?tab=readme-ov-file#pyenv-virtualenv)

Once you have everything installed, you can add additional python version, for example python 3.7.17, you could run the
following commmand:

```
pyenv install 3.7.17

> ```

Reviewer: I think we can skip this part. Contributors mostly shouldn't need to care about different Python versions.

We can recommend using Python 3.7 for development since it's the minimum version we currently support.

Author: Just provided as an example, i could remove it

Note: to switch the default python version, edit the `.python-version` file in the repository root.

## Linting the code

We are using [pre-commit hooks](https://pre-commit.com/#install) to manage oure linters and other auto-generated code.

To install `pre-commit` run the following:
```shell
pip install pre-commit
```

To install all of our pre-commit hooks run:
```shell
pre-commit install
```

If you just want to run a specific hook, like formating your code, you could run the following:
```shell
pre-commit run ruff --all-files --hook-stage pre-push
```

## Testing

### Using pytest

We use the [`pytest`](https://docs.pytest.org/) framework. Tests can be found in `tests/`.

> So you could run test using pytest directly first install, all testing development found in this requirements file:

Reviewer: ```suggestion
All test dependencies should be in `requirements_test.txt`, which you should install before running tests:
```

```shell
pip install -r requirements_test.txt
```

Next, you can install the test depedencies for all test, that are found in this requirements file:
```shell

> pip install -r requirements_dev.txt

Reviewer: Shouldn't `requirements_test.txt` cover all this?

Author: what do you mean?

Reviewer: should we have contributors just do

```bash
pip install -r requirements_test.txt -r requirements_dev.txt
```

in one go? I think the text explaining it doesn't quite make sense. It says that `requirements_dev.txt` is needed for tests, but then it's not clear what is the difference from `requirements_test.txt`.
